### ISSUE TITLE
Unable to create a new app/quickstart at openshift.io (production)

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#1503]
Started seeing this - on manual and automated tests - mid-day (east coast US time) on December 4:

Steps to recreate:
* Create a new space
* Create a new app/quickstart
* Observe this error raised:

![screenshot from 2017-12-04 11-52-11](https://user-images.githubusercontent.com/642621/33564977-dee945c8-d8e9-11e7-80b6-3d81b95fab38.png)

**Comment by corinnekrych** [comment]
@ldimaggi is there anything special on your account, I can't reproduce it:
https://www.dropbox.com/s/p35bpsclaw1wi6h/Screenshot%202017-12-04%2018.44.51.png?dl=0

**Comment by corinnekrych** [comment]
@kbsingh @jfchevrette how can we access prod log?

**Comment by ldimaggi** [comment]
I am seeing this on my personal account and on the account I use for testing. 2 separate accounts.

**Comment by ldimaggi** [comment]
See these 504 forge errors in the Chrome browser console:
```
Failed to load resource: the server responded with a status of 504 (Gateway Time-out)
december401:1 Failed to load https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://openshift.io' is therefore not allowed access. The response had HTTP status code 504.
execute:1 POST https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute 412 (Precondition Failed)
execute:1 POST https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute 412 (Precondition Failed)
```

**Comment by ldimaggi** [comment]
Also seeing these errors:

```
forge.api.openshift.io/forge/commands/fabric8-new-project/execute:1 POST https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute 500 (Internal Server Error)
forge.api.openshift.io/forge/commands/fabric8-new-project/execute:1 POST https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute 412 (Precondition Failed)
forge.api.openshift.io/forge/commands/fabric8-new-project/execute:1 POST https://forge.api.openshift.io/forge/commands/fabric8-new-project/execute 412 (Precondition Failed)
```

**Comment by ldimaggi** [comment]
I am trying this for 2 separate user accounts, on 2 separate physical machines. It fails unless I use incognito mode.

**Comment by ldimaggi** [comment]
This problem seems to be intermittent. Seeing it often - not every time. Is this a service access timing issue?

**Comment by ldimaggi** [comment]
This is failing about 7 out of 10 tries - both manually (on multiple machines) and on jobs running on Centos CI.

**Comment by joshuawilson** [comment]
@gastaldi is something down on the forge side?

**Comment by ldimaggi** [comment]
@jkang is also seeing this issue.

**Comment by gastaldi** [comment]
@joshuawilson I have no idea, this instance is not managed by us yet, but these steps work for me

**Comment by gastaldi** [comment]
Is there something useful in the logs?

**Comment by joshuawilson** [comment]
@gastaldi if you are referring to something managed by James then you should just take it now or we need to figure out who owns it. We need to transition that stuff.

**Comment by gastaldi** [comment]
@joshuawilson yes, that's the main reason why we are unifying our launcher codebase. 

Our backend merge is pratically done, we just need to certify that it does not break anything by running the existing E2E tests against it before we flick the switch. 

Before that happens, I'll be happy to investigate and help identifying the cause.

**Comment by adisakala** [comment]
@gastaldi assigned this to you so you can resolve this accordingly.

**Comment by gastaldi** [comment]
Apparently this error may have been caused by the following error in the pod logs: `java.io.IOException: User limit of inotify watches reached`.

This could be caused the project caching feature in Forge core which listens for changes in the project's pom.xml and evicts the Project cache when that happens (useful when Forge is not running in a webapp). 

A temporary short-term solution would be to increase the inotify watch limit in the running container.

I filled https://issues.jboss.org/browse/FORGE-2769 to provide a way to disable the caching and file monitoring. Once this is fixed in Forge core, I'll release a new Forge version and update the backend accordingly

See log attached: 
[generator-backend-39-0v007.log](https://github.com/openshiftio/openshift.io/files/1531255/generator-backend-39-0v007.log)

**Comment by maxandersen** [comment]
but what could have caused this error to trigger? we are supposedly running immutable images thus  sounds weird if we suddenly have a change here that causes limit of inotify to be reached ?

**Comment by gastaldi** [comment]
I cannot confirm, but maybe a high load of simultaneous users may cause this

**Comment by gastaldi** [comment]
Restarting the pods seems to have solved the issue momentaneously until I provide a definite fix for this.

**Comment by ldimaggi** [comment]
This seems to be working OK now - but - was all that was done just a node restart?

**Comment by gastaldi** [comment]
Yes. I wonder if the health check probe can be enhanced to detect such failures

**Comment by joshuawilson** [comment]
@kbsingh do you own the health check?

**Comment by bartoszmajsak** [comment]
I believe healtcheck should be owned by respective service teams and SD team can advise/define what should be the part of it.

**Comment by maxandersen** [comment]
Anyone actually know what controls forge deployment these days ? is it https://github.com/openshiftio/saas-openshiftio/tree/master/homeless-templates ? 

@vpavlin can you help clarify ?

**Comment by gastaldi** [comment]
Now that production is running smooth due to the node restart, is this issue still considered SEV-1?

**Comment by gastaldi** [comment]
I created https://github.com/fabric8io/fabric8-generator/pull/81 and https://github.com/fabric8io/generator-backend/pull/77 to fix the inotify issues. Can someone review and run a full E2E test once it's merged?

Thanks

**Comment by maxandersen** [comment]
@rawlingsj @jstrachan can one of you help @gastaldi here ?

**Comment by rawlingsj** [comment]
I think @corinnekrych is picking up work on forge AFAIK and is best placed to review etc.

**Comment by vpavlin** [comment]
> Anyone actually know what controls forge deployment these days ? is it https://github.com/openshiftio/saas-openshiftio/tree/master/homeless-templates ?

Forge is built in/deployed from GKE, but we still wanted to track the generated template in saas-openshiftio repository, so we added homeless-templates and a service yaml pointing to that dir.

**Comment by corinnekrych** [comment]
@vpavlin talking with @gastaldi it seems this template https://github.com/openshiftio/saas-openshiftio/blob/master/homeless-templates/generator-backend.yaml is not used but this one is used to deploy to prod: https://github.com/fabric8io/generator-backend/blob/master/src/main/fabric8/deployment.yml#L16
right?

**Comment by vpavlin** [comment]
Correct

**Comment by gastaldi** [comment]
When is https://github.com/fabric8io/generator-backend/pull/77 going to be merged? @corinnekrych do you have the necessary privileges to do so?

**Comment by corinnekrych** [comment]
@gastaldi no I don't have merge rights
cc @jstrachan @joshuawilson

**Comment by joshuawilson** [comment]
Get someone to review it and approve then I will merge it.

**Comment by ppitonak** [comment]
@gastaldi what is the status of this issue?

**Comment by gastaldi** [comment]
It can be safely closed.

It is no longer happening and it should hopefully be permanently fixed when the generator-backend service is replaced with launcher-backend

**Comment by ldimaggi** [comment]
Closing - the issue is not being seen.

**Comment by pbergene** [comment]
Just got this today, quickstarts crashing  "Caused by: java.io.IOException: User limit of inotify watches reached"  Front end showed the "repository X already exists".  Lost the full stacktrace from my clipboard, but available in kibana (deployment #40).

Restart of generator-backend removed the issue.

**Comment by gastaldi** [comment]
We need to replace the existing Forge service with our launcher-backend ASAP. Who in the SD Team can help us with that?

**Comment by joshuawilson** [comment]
@gastaldi do you need this open and should the Severity level be raised? Or do you need a new issue?

**Comment by pbergene** [comment]
It's a little down to process really, do we close an issue when a fix is on it's ready or when it has made it to production.  It thought it was a bit confusing when I hit what looked like this issue in production and it had been closed.

**Comment by joshuawilson** [comment]
If the fix is linked via the PR then it gets closed automatically when the PR gets merged. However, it has been my practice to close things when they are in prod if it wasn't auto-closed.
